x-pack/plugins/security_solution/server/lib/entity_analytics/entity_store/entity_store_data_client.ts

```diff
@@ -131,6 +157,42 @@ export class EntityStoreDataClient {
     });
   }
 
+  private async getEngineComponentsState(
```

Comment:
not for this PR, but I think this Data Client is getting too big. Perhaps we need to introduce `EntityStoreServices`, similar to how we do it in the Risk Engine? We could then have a service grouping together stuff pertaining to specific features/use cases

@hop-dev any thoughts on this?

Reply:
I agree 100%!

Reply:
I agree, lets nip this in the bud before it gets to 1000 lines :) I have created https://github.com/elastic/security-team/issues/11293